src/Compilers/CSharp/Portable/Binder/BinderFactory.BinderFactoryVisitor.cs

```diff
@@ -339,7 +340,8 @@ private Binder VisitOperatorOrConversionDeclaration(BaseMethodDeclarationSyntax
                         resultBinder = new InMethodBinder(method, resultBinder);
                     }
 
-                    resultBinder = resultBinder.WithUnsafeRegionIfNecessary(parent.Modifiers);
+                    resultBinder = resultBinder.WithUnsafeRegionIfNecessary(parent.Modifiers,
+                        isIteratorBody: inBody && method?.IsIterator == true);
```

Comment:
Do we have tests with operators?

Reply:
Yes, `UnsafeContext_Operator` and `UnsafeContext_Operator_Iterator`.